src/extensions/as-legacy-globals-for-extension-api/legacy-global-di-for-extension-api.ts

```diff
@@ -4,10 +4,29 @@
  */
 import type { DiContainer } from "@ogre-tools/injectable";
 
-let legacyGlobalDi: DiContainer;
+const legacyGlobalDis = new Map<Environments, DiContainer>();
 
-export const setLegacyGlobalDiForExtensionApi = (di: DiContainer) => {
-  legacyGlobalDi = di;
+export enum Environments {
+  renderer,
+  main,
+}
+
+export const setLegacyGlobalDiForExtensionApi = (
+  di: DiContainer,
+  environment: Environments,
+) => {
+  legacyGlobalDis.set(environment, di);
+};
+
+export const getLegacyGlobalDiForExtensionApi = () => {
+  const globalDis = [...legacyGlobalDis.values()];
+
+  if (globalDis.length > 1) {
+    throw new Error("Tried to get DI container using legacy globals where there is multiple containers available.");
+  }
```

Comment:
Yes, this is there to prevent hard to find "why is this test green or red" -moments. Can't occur in production since both  there is always only one DI, but in unit tests this can happen very easily at the moment. 

Of course, in long run we are removing these "legacy-global" -functions for good, so it will get removed as well.